dist/commonjs/router/router.js

```diff
@@ -55,7 +63,7 @@ function getTransitionByIntent(intent, isIntermediate) {
     }
 
     // No-op. No need to create a new transition.
-    return new Transition(this);
+    return this.activeTransition || new Transition(this);
```

Comment:
You may already know this, but with some git magic you could fix this. If you...

1. checked out a branch from master
2. built and PR'd that
3. came back here, reset the changes to dist
3. rebased this PR off that other PR

...it would make few a cleaner diff here and more atomic commit history.